parserasgeo/features/cross_section.py

```diff
@@ -514,6 +520,76 @@ def cut_line_ratio(self):
             last_pt = pt
 
         return cl_length/length
+    
+    def define_channel_n(self):
+        """
+        Checks if the bank stations are represented in Manning's n (i.e. if Manning's n is defined at the statiosn that correspond with the bank stations)
+        If they are not defined at the bank stations, add Manning's n values at those points.
+        """
+        
+        left_bank = self.bank_sta.left
+        right_bank = self.bank_sta.right
+        banks = [left_bank, right_bank]
+        
+        manning_tuples = self.mannings_n.values
+        manning_stations = [x[0] for x in manning_tuples]
+        
+        for bank in banks:
+            # if the bank station is not in the list, grab the Manning's n to the left
+            if bank not in manning_stations:
+                left_manning = [x[1] for x in manning_tuples if x[0] < bank]
+                left_index = len(left_manning)
+                left_manning = left_manning[-1]
+                manning_insert = (bank, left_manning, 0)
+                manning_tuples.insert(left_index, manning_insert)
+        
+        # do not pull the n-value at the right bank because that defines the n-values to the right of the bank
+        self.channel_n = [x for x in manning_tuples if (x[0] >= banks[0] and x[0] < banks[1])]        
+            
+    def alter_channel_n(self, scalar):
+        """
+        Alters the channel n-values by a scaling factor
+        """
+        
+        if self.channel_n is not None:
```

Comment:
I've been trying to define public attributes in __init__(), and setting them to None if they are not set in __init__() with a comment about what they are for and where they get defined. This provides a degree of documentation about which attributes are available in which classes.